### PR TITLE
MULE-16274: HTTP service client cannot do gzip response decompression

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

File: modules/http-api/src/main/java/org/mule/runtime/http/api/client/HttpClientConfiguration.java [hunk 1]
```diff
@@ -79,6 +85,10 @@ public String getName() {
     return name;
   }
 
+  public Optional<Boolean> isDecompress() {
```
Comment (elrodro83) [inline modules/http-api/src/main/java/org/mule/runtime/http/api/client/HttpClientConfiguration.java]: Why an optional? Wgy not just `boolean`?

**Comment by afelisatti** [comment]
--validate

**Comment by afelisatti** [comment]
--validate

**Comment by afelisatti** [comment]
--validate

**Comment by afelisatti** [comment]
--validate

**Pull request merged**